request.hpp

```diff
@@ -3,23 +3,41 @@
 #ifndef HTTP_REQUEST_HPP
 #define HTTP_REQUEST_HPP
 #include "header.hpp"
+#include "request_parser.hpp"
 #include <string>
 #include <vector>
+#include <memory>
 
 namespace http {
 namespace server {
 
+struct request_parser_output;
 
-/// A request received from a client.
-struct request
-{
-  std::string method;
-  std::string uri;
-  int http_version_major;
-  int http_version_minor;
-  std::vector<header> headers;
-  std::string content;
-  long unsigned int bytes;
+class Request {
+ public:
+	static std::unique_ptr<Request> Parse(const std::string& raw_request);
+
+	std::string raw_request() const;
+	std::string method() const;
+	std::string uri() const;
+	std::string version() const;
+
+	//TODO: check why this is written like this
+	using Headers = std::vector<std::pair<std::string, std::string>>;
+	Headers headers() const;
+
```

Comment:
yeah it just defines the function with return type std::vector<std::pair<std::string, std::string>> but i was not sure why it was written like this in 2 lines in the common api